lib/core/request.js

```diff
@@ -141,8 +141,11 @@ class Request {
 
     this.path = query ? serializePathWithQuery(path, query) : path
 
+    // TODO: shall we maybe standardise it to an URL object?
     this.origin = origin
 
+    this.protocol = new URL(origin).protocol
```

Comment:
Why tho? Either the origin starts with 'https:' or with 'http://'. We can add a util function to `/lib/core/util.js` next to `isHttpOrHttpsPrefixed`

```js
function getProtocolFromUrl (urlString) {
  if (
    urlString[0] === 'h' &&
    urlString[1] === 't' &&
    urlString[2] === 't' &&
    urlString[3] === 'p'
    ) {
      switch (urlString[4]) {
        case ':':
          return 'http:'
        case 's':
          if (urlString[5] === ':') {
            return 'https:'
          }
      }
   }
  // fallback if none of the usual suspects
  return urlString.slice(0, urlString.indexOf(':') + 1)
} 
```